### PR TITLE
Add tiller resource for chart and chartconfig controllers

### PR DESCRIPTION
**Pull request by rossf7**
As we discussed in sig-operator sync moves all the `EnsureTillerInstalled` calls to the tiller resource which is really simple.

@xh3b4sd Thx really cool idea!

### CODE REVIEW

File: service/controller/chart/v1/resource/release/create.go [hunk 1]
```diff
@@ -39,11 +39,6 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 			}
 		}()
 
-		err = r.helmClient.EnsureTillerInstalled(ctx)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-
```
Comment (rossf7) [inline service/controller/chart/v1/resource/release/create.go]: All these calls move to the tiller resource which is executed first.

File: LICENSE [hunk 2]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2016 - 2019 GiantSwarm GmbH
+   Copyright 2016 - 2019 Giant Swarm GmbH
```
Comment (rossf7) [inline LICENSE]: Fixing the damn license.

File: service/controller/chart/v1/resource/tiller/create.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package tiller
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+)
+
```
Comment (rossf7) [inline service/controller/chart/v1/resource/tiller/create.go]: Resource is super simple.

File: service/controller/chart/v1/resource_set.go [hunk 4]
```diff
@@ -93,10 +94,23 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
...
+			return nil, microerror.Mask(err)
+		}
+	}
+
 	resources := []controller.Resource{
-		// statusResource executed first so the status is set for existing CRs.
-		statusResource,
+		tillerResource,
 		releaseResource,
+		statusResource,
```
Comment (rossf7) [inline service/controller/chart/v1/resource_set.go]: @xh3b4sd Note I moved the status resource to last.

I had problems when testing locally. It also means the status resource reflects any changes made by the release resource.

File: service/controller/chartconfig/v5/resource/chart/create.go [hunk 5]
```diff
@@ -38,11 +38,6 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 			}
 		}()
 
-		err = r.helmClient.EnsureTillerInstalled(ctx)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-
```
Comment (rossf7) [inline service/controller/chartconfig/v5/resource/chart/create.go]: Same change for chartconfig controller.

File: service/controller/chartconfig/v5/resource/tiller/create.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "ensuring tiller is installed")
+
+	err := r.helmClient.EnsureTillerInstalled(ctx)
```
Comment (xh3b4sd) [inline service/controller/chartconfig/v5/resource/tiller/create.go]: Calling it once and for all is already a huge improvement. We still have to improve the internals of this thing as the PMs show.

File: service/controller/chart/v1/resource/tiller/delete.go [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package tiller
+
+import (
+	"context"
+)
+
+// EnsureDeleted is not implemented for the tiller resource.
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	return nil
```
Comment (xh3b4sd) [inline service/controller/chart/v1/resource/tiller/delete.go]: I think we should ensure the installation here as well. Once use case would be an upgrade necessary to perform the action at all. Another would be a tenant cluster creation that is not completed but the tenant cluster got requested to be deleted. Then some code could fail down the road during deletion.

File: service/controller/chartconfig/v5/resource/tiller/create.go (hunk 6 above)
Comment (rossf7) [reply service/controller/chartconfig/v5/resource/tiller/create.go]: Totally, we still need to improve helmclient. I just want to get chart-operator "done done" for app-catalog. So I wanted to get this in.

File: service/controller/chart/v1/resource/tiller/delete.go (hunk 7 above)
Comment (rossf7) [reply service/controller/chart/v1/resource/tiller/delete.go]: Good call. I implemented both methods and moved them to resource.go.

**Comment by rossf7** [comment]
@xh3b4sd Should be good now. PTAL.

**Pull request merged**